training/training.module

```diff
@@ -0,0 +1,505 @@
+    'access callback' => TRUE,
+    'file' => 'training.pages.inc',
+  );
+  $menu['training/form'] = array(
+    'title' => 'Form Example',
+    'description' => 'This page is Form Example',
+    'page callback' => 'training_example_form_callback',
+    'page arguments' => array('training_google_form'),
+    'access callback' => TRUE,
+    'file' => 'training.pages.inc',
+  );
+
+  return $menu;
+}
+
+/**
+ * Check access for page Custom Access Menu Item.
+ *
+ * @param string $role_name
+ *   The role required for access.
+ *
+ * @return bool
+ *   True if the acting user has the role specified.
+ */
+function training_custom_access($role_name) {
+  global $user;
+
+  return in_array($role_name, $user->roles);
+}
+
+/**
+ * Displays a specific string, depending on a number in the address bar.
+ *
+ * @param int $arg
+ *   Integer argument in the address bar.
+ *
+ * @return string
+ *   A string that corresponds to an integer in the address bar, or NULL if there is no match.
+ */
+function digit_load($arg) {
+  switch ($arg) {
+    case 1:
+      $arg = 'one';
+      break;
+    case 2:
+      $arg = 'two';
+      break;
+    case 3:
+      $arg = 'three';
+      break;
+    case 99:
+      $arg = 'Almost there';
+      break;
+    default:
+      $arg = NULL;
+  }
+
+  return $arg;
+}
+
+/**
+ * Title callback to rename the title dynamically.
+ *
+ * @return string
+ *   A string that contents dynamic title.
+ */
+function training_menu_title_callback() {
+  global $user;
+  $user_name = isset($user->name) ? $user->name : 'anonymous';
+
+  return 'Dynamic title: username = ' . $user_name;
+}
+
+/**
+ * Implements hook_permission().
+ */
+ function training_permission() {
+  $permission = array(
+    'train' => array(
+      'title' => 'Access protected menu example',
+      'description' => 'Use this to control access to page Permissioned Menu Item of training module',
+    ),
+    'train_theme' => array(
+      'title' => 'Access protected content',
+      'description' => 'Use this to control access to page Access Theme of training module',
+    ),
+  );
+
+  return $permission;
+}
+
+/**
+ * Creates example form.
+ *
+ * @param array $form
+ *   Array of form elements.
+ * @param array $form_state
+ *   Array of form parameters.
+ *
+ * @return array
+ *   Array of form elements.
+ */
+function training_google_form($form, &$form_state) {
+  $form = array();
+  $form['group'] = array(
+    '#type' => 'fieldset',
+    '#title' => 'group',
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+  );
+  $form['group']['text'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title',
+  );
+  $form['group']['text2'] = array(
+    '#type' => 'textfield',
+    '#title' => 'title2',
+    '#required' => TRUE,
+  );
+  $form['#attached']['css'] = array(
+    //Path to the css file
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('ok'),
+  );
+
+  return $form;
+}
+
+/**
```

Comment:
Move all hooks to the beginning of .module file
All other functions and callbacks go after hooks